File: AI/MC.java
import java.util.Scanner;

public class MC {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int lM = 3; 
        int lC = 3; 
        int rM = 0; 
        int rC = 0; 
        int k = 0; 

            while (true) {

                if ((rM + rC) == 6) {
                    System.out.println("Win..!!!");
                    break;
                }

                while (true) {
                    System.out.println("\n===================================");
                    System.out.println("LM : "+lM+" & LC : "+lC+" && RM : "+rM+" & RC : "+rC);
                    System.out.println("===================================\n");

                
                    System.out.println("Left to Right : ");

                    System.out.print("Enter number of Missionaries : ");
                    int uM = scanner.nextInt();
                    System.out.print("Enter number of Cannibals : ");
                    int uC = scanner.nextInt();

                    if (uM == 0 && uC == 0) {
                        System.out.println("Empty travel not possible");
                        System.out.println("Re-enter : ");
                    } else if ((uM + uC <= 2) && (lM - uM >= 0) && (lC - uC >= 0)) {
                        lM -= uM;
                        lC -= uC;
                        rM += uM;
                        rC += uC;
                        break;
                    } else {
                        System.out.println("Wrong input re-enter : ");
                    }
                }

                if ((rM + rC) == 6) {
                    System.out.println("Win..!!!");
                    break;
                }

                if (isInvalidState(lM, lC) || isInvalidState(rM, rC)) {
                    System.out.println("Cannibals eat missionaries...");
                    break;
                }

                while (true) {

                    System.out.println("\n===================================");
                    System.out.println("LM : "+lM+" & LC : "+lC+" && RM : "+rM+" & RC : "+rC);
                    System.out.println("===================================\n");

                    System.out.println("Right To Left : ");
                    System.out.print("Enter number of Missionaries : ");
                    int userM = scanner.nextInt();
                    System.out.print("Enter number of Cannibals : ");
                    int userC = scanner.nextInt();

                    if (userM == 0 && userC == 0) {
                        System.out.println("Empty travel not possible");
                        System.out.println("Re-enter : ");
                    } else if ((userM + userC <= 2) && (rM - userM >= 0) && (rC - userC >= 0)) {
                        lM += userM;
                        lC += userC;
                        rM -= userM;
                        rC -= userC;
                        break; 
                    } else {
                        System.out.println("Wrong input re-enter : ");
                    }
                }

                if (isInvalidState(lM, lC) || isInvalidState(rM, rC)) {
                    System.out.println("Cannibals eat missionaries:");
                    break;
                }
            }
    }

    private static boolean isInvalidState(int m, int c) {
        return (c > m && m > 0); 
    }
}
    

File: AI/mc.py
def is_invalid_state(m, c):
    return c > m > 0


def main():
    lM, lC, rM, rC = 3, 3, 0, 0

    while True:
        if (rM + rC) == 6:
            print("Win..!!!")
            break

        while True:
            print("\n===================================")
            print(f"LM : {lM} & LC : {lC} && RM : {rM} & RC : {rC}")
            print("===================================\n")

            print("Left to Right : ")
            uM = int(input("Enter number of Missionaries: "))
            uC = int(input("Enter number of Cannibals: "))

            if uM == 0 and uC == 0:
                print("Empty travel not possible")
                print("Re-enter : ")
            elif (uM + uC <= 2) and (lM - uM >= 0) and (lC - uC >= 0):
                lM -= uM
                lC -= uC
                rM += uM
                rC += uC
                break
            else:
                print("Wrong input re-enter : ")

        if (rM + rC) == 6:
            print("Win..!!!")
            break

        if is_invalid_state(lM, lC) or is_invalid_state(rM, rC):
            print("Cannibals eat missionaries...")
            break

        while True:
            print("\n===================================")
            print(f"LM : {lM} & LC : {lC} && RM : {rM} & RC : {rC}")
            print("===================================\n")

            print("Right To Left : ")
            userM = int(input("Enter number of Missionaries: "))
            userC = int(input("Enter number of Cannibals: "))

            if userM == 0 and userC == 0:
                print("Empty travel not possible")
                print("Re-enter : ")
            elif (userM + userC <= 2) and (rM - userM >= 0) and (rC - userC >= 0):
                lM += userM
                lC += userC
                rM -= userM
                rC -= userC
                break
            else:
                print("Wrong input re-enter : ")

        if is_invalid_state(lM, lC) or is_invalid_state(rM, rC):
            print("Cannibals eat missionaries...")
            break


if __name__ == "__main__":
    main()
